addons/portal/controllers/mail.py

```diff
@@ -183,13 +184,19 @@ def portal_message_fetch(self, res_model, res_id, domain=False, limit=10, offset
                 raise Forbidden()
             # Non-employee see only messages with not internal subtype (aka, no internal logs)
             if not request.env['res.users'].has_group('base.group_user'):
-                domain = expression.AND([Message._non_employee_message_domain(), domain])
+                domain = expression.AND([Message._get_search_domain_share(), domain])
             Message = request.env['mail.message'].sudo()
         return {
             'messages': Message.search(domain, limit=limit, offset=offset).portal_message_format(),
             'message_count': Message.search_count(domain)
         }
 
+    @http.route(['/mail/toggle_internal'], type='json', auth="user", website=True)
+    def portal_message_toggle_internal(self, message_id):
```

Comment:
I don't like toggles, I would prefer explicitly passing the target value. Avoids issues when double clicking on the button, or multiple people wanting to change the value at the same time, etc.

Reply:
Ok, I will set as an update of is_internal + value explicitely given.